staging/src/k8s.io/apiserver/pkg/endpoints/installer.go

```diff
@@ -1070,7 +1070,7 @@ func splitSubresource(path string) (string, string, error) {
 
 // GetArticleForNoun returns the article needed for the given noun.
 func GetArticleForNoun(noun string, padding string) string {
-	if noun[len(noun)-2:] != "ss" && noun[len(noun)-1:] == "s" {
+	if len(noun) >= 2 && noun[len(noun)-2:] != "ss" && noun[len(noun)-1:] == "s" {
```

Comment:
this would also be simpler using strings.HasSuffix than manual slicing

Reply:
Thanks @liggitt 
Done!